buckets/bucket_impl_tests.go

```diff
@@ -63,7 +63,10 @@ func TestGC(t *testing.T, factory quotaservice.BucketFactory, impl string) {
 	helpers.CheckError(t, config.AddNamespace(cfg, nsCfg))
 
 	eventsEmitter := &quotaservice.MockEmitter{Events: make(chan events.Event, 100)}
-	container := quotaservice.NewBucketContainer(cfg, factory, eventsEmitter, quotaservice.NewReaperConfigForTests())
+	container := quotaservice.NewBucketContainer(factory, eventsEmitter, quotaservice.NewReaperConfigForTests())
+	container.Lock()
```

Comment:
We technically don't but it's good practice that we do. 

+1 to your suggestion re: `Init()`.